uberfire-extensions/uberfire-commons-editor/uberfire-commons-editor-client/src/main/java/org/uberfire/ext/editor/commons/client/BaseEditor.java

```diff
@@ -235,6 +246,60 @@ protected void makeMenuBar() {
         }
     }
 
+    protected Command getSaveAndRename() {
+
+        return getSaveAndRenameCommandBuilder()
+                .addPathSupplier(getPathSupplier())
+                .addValidator(getRenameValidator())
+                .addValidator(getSaveValidator())
+                .addRenameService(getSaveAndRenameServiceCaller())
+                .addMetadataSupplier(getMetadataSupplier())
+                .addContentSupplier(getContentSupplier())
+                .addIsDirtySupplier(isDirtySupplier())
+                .addSuccessCallback(onSuccess())
+                .build();
+    }
+
+    Supplier<Boolean> getSaveValidator() {
+
+        return () -> {
+
+            if (isReadOnly && versionRecordManager.isCurrentLatest()) {
+                baseView.alertReadOnly();
+                return false;
+            } else if (isReadOnly && !versionRecordManager.isCurrentLatest()) {
+                versionRecordManager.restoreToCurrentVersion();
+                return false;
+            }
+
+            if (concurrentUpdateSessionInfo != null) {
+                showConcurrentUpdatePopup();
+                return false;
+            } else {
+                return true;
+            }
+        };
+    }
+
+    protected ParameterizedCommand<Path> onSuccess() {
+
+        return (path) -> {
+
+            final T content = getContentSupplier().get();
+            final int originalHash = content.hashCode();
+
+            setOriginalHash(originalHash);
```

Comment:
Presumably this is called after a save/save-and-rename is successful. Should you also set the ```metadataOriginalHash```?

Reply:
Done.